### PR TITLE
Ft/s3 c 1156 get bucket lifecycle

### PR DESCRIPTION
**Pull request by dora-korpar**
Implements getBucketLifecycle API
Depends on https://github.com/scality/Arsenal/pull/397

https://github.com/scality/S3/pull/1061 must be merged first

### CODE REVIEW

File: lib/api/bucketGetLifecycle.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        if (!lifecycleConfig) {
+            log.debug('error processing request', {
+                error: errors.NoSuchLifecycleConfiguration,
+                method: 'bucketGetLifecycle',
+            });
+            return callback(errors.NoSuchLifecycleConfiguration, null,
+                corsHeaders);
+        }
+        const xml = LifecycleConfiguration.getConfigXml(lifecycleConfig);
+        pushMetric('getBucketReplication', log, {
```
Comment (bennettbuchanan) [inline lib/api/bucketGetLifecycle.js]: ~getBucketReplication~ getBucketLifecycle

File: tests/functional/aws-node-sdk/test/bucket/getBucketLifecycle.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+                Bucket: bucket,
+                LifecycleConfiguration: lifecycleConfig,
+            }, err => {
+                assert.equal(err, null, `Err putting lifecycle config: ${err}`);
+                s3.getBucketLifecycleConfiguration({ Bucket: bucket },
+                (err, res) => {
+                    assert.equal(err, null, 'Error getting lifecycle config: ' +
+                        `${err}`);
+                    expectedKeys.forEach(key =>
+                        assert(res.Rules[0][key], true));
```
Comment (bennettbuchanan) [inline tests/functional/aws-node-sdk/test/bucket/getBucketLifecycle.js]: A bit confused by this as the second argument in `assert` would be a message. Maybe we can instead do a deepStrictEqual on the `Rules[0]` object?

File: tests/unit/api/bucketGetLifecycle.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+const log = new DummyRequestLogger();
+const authInfo = makeAuthInfo('accessKey1');
+const bucketName = 'bucketname';
+const testBucketPutRequest = {
+    bucketName,
+    headers: { host: `${bucketName}.s3.amazonaws.com` },
+    url: '/',
+};
+
+function _makeLifecycleRequest(xml) {
```
Comment (bennettbuchanan) [inline tests/unit/api/bucketGetLifecycle.js]: Looks like these private test methods are duplicates of the utility methods in tests/unit/utils/lifecycleHelpers.js. (Sorry in advance if I'm reviewing this too soon 😸.)

Comment (dora-korpar) [reply tests/unit/api/bucketGetLifecycle.js]: Yeah, they basically are. Next step was going to be to pull them out :)

**Comment by bennettbuchanan** [comment]
@ironman-machine r+

**Comment by bennettbuchanan** [comment]
The "r+" command was successful but merging failed due to pending ci/circleci_enterprise tests. Those tests have now completed successfully. Merging manually.

**Pull request merged**